### PR TITLE
feat: add support library

### PR DESCRIPTION
**Pull request by alresarena2021**
feat: add support library

### CODE REVIEW

File: packages/support/README.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+app.listen(8080);
+```
+
+`Add support message to notion`
+
+```json
+// POST localhost:8080/support/messages
+
+{
+  "customer": "alres arena",
```
Comment (rogermadjos) [inline packages/support/README.md:30]: hmm, I guess we should just put the `customerName` inside `details`
```
{
  emailAddress: string;
  message: string;
  details?: Record<string, unknown>
}
```

**Review by rogermadjos — CHANGES_REQUESTED** [review]

File: packages/support/README.md (hunk 1 above)
Comment (alresarena2021) [reply packages/support/README.md]: fixed

File: packages/support/README.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+```json
+// POST localhost:8080/support/messages
+
+{
+  "emailAddress": "alres.arena2019@gmail.com",
+  "message": "this is a test message",
+  "details": {
+    "customerName": "alres arena",
+    "name": "alres arena",
```
Comment (rogermadjos) [inline packages/support/README.md:34]: remove this please

File: packages/support/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+version: '3.9'
```
Comment (rogermadjos) [inline packages/support/docker-compose.yml:1]: why do you need a compose file?

File: packages/support/package.json [hunk 4]
```diff
@@ -0,0 +1,75 @@
+{
+  "name": "@highoutput/support",
+  "version": "0.1.0",
```
Comment (rogermadjos) [inline packages/support/package.json:3]: please set the version to `0.0.1`

File: packages/support/package.json [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    "express": "^4.18.1",
+    "get-port": "^5.1.1",
+    "jest": "^28.1.0",
+    "jsonwebtoken": "^8.5.1",
+    "supertest": "^6.2.3",
+    "ts-jest": "^28.0.2",
+    "ts-node": "^10.7.0",
+    "ts-node-dev": "^1.1.8"
+  },
+  "dependencies": {
```
Comment (rogermadjos) [inline packages/support/package.json:51]: please remove unnecessary dependencies

File: packages/support/src/support-server.ts [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+  }
+
+  private schema = {
+    type: 'object',
+    properties: {
+      emailAddress: { type: 'string' },
+      message: { type: 'string' },
+    },
+    required: ['emailAddress', 'message'],
+    additionalProperties: true,
```
Comment (rogermadjos) [inline packages/support/src/support-server.ts:34]: set this to `false`

File: packages/support/src/support-server.ts [hunk 7]
```diff
@@ -0,0 +1,216 @@
...
+    const ajv = new Ajv();
+
+    this.validator = ajv.compile(this.schema);
+  }
+
+  private schema = {
+    type: 'object',
+    properties: {
+      emailAddress: { type: 'string' },
+      message: { type: 'string' },
```
Comment (rogermadjos) [inline packages/support/src/support-server.ts:31]: add the `details` property

**Review by rogermadjos — CHANGES_REQUESTED** [review]

File: packages/support/README.md (hunk 2 above)
Comment (alresarena2021) [reply packages/support/README.md]: fixed

File: packages/support/docker-compose.yml (hunk 3 above)
Comment (alresarena2021) [reply packages/support/docker-compose.yml]: fixed

File: packages/support/package.json (hunk 4 above)
Comment (alresarena2021) [reply packages/support/package.json]: fixed

File: packages/support/package.json (hunk 5 above)
Comment (alresarena2021) [reply packages/support/package.json:48]: fixed

File: packages/support/src/support-server.ts (hunk 6 above)
Comment (alresarena2021) [reply packages/support/src/support-server.ts]: fixed

File: packages/support/src/support-server.ts (hunk 7 above)
Comment (alresarena2021) [reply packages/support/src/support-server.ts:31]: fixed

**Review by rogermadjos — APPROVED** [review]

**Pull request merged**